ibis/backends/duckdb/__init__.py

```diff
@@ -1598,6 +1599,12 @@ def _register_in_memory_table(self, op: ops.InMemoryTable) -> None:
             # only register if we haven't already done so
             self.con.register(name, op.data.to_pyarrow(op.schema))
 
+            # if we don't aggressively unregister tables duckdb will keep a
+            # reference to every memtable ever registered, even if there's no
+            # way for a user to access the operation anymore, resulting in a
+            # memory leak
+            weakref.finalize(op, self.con.unregister, name)
```

Comment:
Ultimately we probably need to do similar things for other local backends, and perhaps even non-local ones to ensure that we don't balloon storage.

Reply:
The non-local ones at least shouldn't fill local memory, but they would result in an excess of unreachable temp tables. Something to fixup for later (I'll open an issue).

Reply:
Do you want to open an issue, if you haven't already, to track this for other local backends?

Reply:
Already done: https://github.com/ibis-project/ibis/issues/10044